pkg/scheduler/framework/interface.go

```diff
@@ -617,7 +617,11 @@ type PostFilterResult struct {
 type PodNominator interface {
 	// AddNominatedPod adds the given pod to the nominator or
 	// updates it if it already exists.
-	AddNominatedPod(pod *PodInfo, nodeName string)
+	// Notes for the semantics of `nodeName`:
+	// - if it's nil, the backing nominator aborts if the pod doesn't carry a nominatedNodeName; otherwise
+	//   sets the carried nominatedNodeName.
+	// - if it's non-nil, the internal nominating bits will be updated with the given `nodeName`.
+	AddNominatedPod(pod *PodInfo, nodeName *string)
```

Comment:
Overloading the parameter nodeName to signal more than simply the nodeName is problematic, I suggest making the interface explicit. Here are some options:

1) Is there really a legitimate case where the nodeName is different from the one set in the passed pod if one is set? If so, I think we need to revisit that because I don't think there should be one.

2) If we fail with option 1, I think this function should have consistent semantics and always adds pod to the passed nodeName, no exceptions and fail on an empty `nodeName`. The caller of this function can do 
use a wrapper function as follows (not part of this interface):

```
func AddNominatedPodIfSet(pod) {
 if pi.Pod.Status.NominatedNodeName != "" {
  AddNominatedPod(pod, pi.Pod.Status.NominatedNodeName)
 }
}
```

Reply:
Re 1: yes, it can be empty in recordSchedulingFailure() to clear the entry, or UpdateNominatedPod() which reveals a special case that right after a preemptor is marked as nominated in memory, and at the same time a podUpdate event (doesn't carry nominatedNodeName) comes in.

Re 2: the wrapper can help refactor the code for better readability. Thanks.